src/app/content/components/ContentExcerpt.tsx

```diff
@@ -1,24 +1,44 @@
 import React from 'react';
+import { useSelector } from 'react-redux';
 import styled from 'styled-components/macro';
 import { bodyCopyRegularStyle } from '../../components/Typography';
-import addTargetBlankToLinks from '../utils/addTargetBlankToLinks';
+import { book } from '../selectors';
+import { Page } from '../types';
+import addTargetBlankAndFixRelativeURL from '../utils/addTargetBlankAndFixRelativeURL';
+import { findArchiveTreeNodeById } from '../utils/archiveTreeUtils';
+import { getBookPageUrlAndParams } from '../utils/urlUtils';
 
 interface Props {
   content: string;
   className: string;
+  sourcePageId: string;
 }
 
 // tslint:disable-next-line:variable-name
 const ContentExcerpt = styled((props: Props) => {
   const {
     content,
     className,
+    sourcePageId,
     ...excerptProps
   } = props;
 
+  const currentBook = useSelector(book);
+
+  if (!currentBook) {
+    return null;
+  }
+
+  const sourcePage = findArchiveTreeNodeById(currentBook.tree, sourcePageId) as unknown as Page;
+
+  const excerptSource = getBookPageUrlAndParams(
+    currentBook,
+    sourcePage
+  );
+
   const fixedContent = React.useMemo(
-    () => addTargetBlankToLinks(props.content),
-    [props.content]
+    () => addTargetBlankAndFixRelativeURL(props.content, excerptSource.url),
```

Comment:
don't combine these into one function. leave the old one the way it was and create a new one for fixing the relative urls.

https://en.wikipedia.org/wiki/Single-responsibility_principle